Examples for tslint-documentation

Shinigami92 opened the pull request:
#### PR checklist

- [x] Addresses an existing issue: #3601
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [x] Documentation update

#### Overview of change:


#### Is there anything you'd like reviewers to focus on?

I'm not an english native speaker, pls check my sentences :)

#### CHANGELOG.md entry:

[enhancement] Documentation examples for following rules: `arrow-return-shorthand`, `class-name`, `no-any`, `no-empty-interface`, `no-sparse-arrays`, `no-unnecessary-callback-wrapper`, `object-literal-sort-keys`, `one-variable-per-declaration`, `only-arrow-functions`, `prefer-template`, `radix`, `switch-default`, `typedef`, `use-isnan`

reduckted committed:
completed-docs: corrected example config for tags (#3578)

listepo-alterpost committed:
TSLint rules for RxJS (#3599)

mateuszwitkowski committed:
fix no unnecessary space before parameter from arrow-parens fixer (#3618)

brunolemos committed:
[new-rule-option] [object-literal-sort-key] Add shorthand-first option (#3607)

Commit:
[docs] [ban-comma-operator] metadata updates (#3612)

ajafff committed:
fix condition for deprecation of typeof-compare (#3429)

ajafff committed:
refactor: removed 'export = Linter` in linter.ts (#3438)

[no-log] This doesn't change our public API.

ajafff committed:
no-unsafe-any: allow object spread (#3439)

[bugfix] `no-unsafe-any` allows spreading an `any`-value into an object

ajafff committed:
callable-types: add parens inside ArrayType (#3440)

[bugfix] `callable-types` adds parentheses when fixing a type literal inside an array type

ajafff committed:
stylish-formatter: make filename clickable (#3491)

Ref: https://github.com/palantir/tslint/issues/3460#issuecomment-342396889

ajafff committed:
launch.json: show stdout in debug console (#3464)

Fixes: #3457
Supersedes: #3458

ajafff committed:
no-unnecessary-type-assertion: extract checking of NonNullExpression (#3468)

ajafff committed:
Add test for no-unnecesary-type-assertion (#3535)

ajafff committed:
Remove test for plain test output (#3407)

ajafff committed:
no-unsafe-any: correctly handle types and type-only namespaces (#3487)

ajafff committed:
strict-type-predicates: allow comparing typeof with expression (#3542)

ajafff committed:
Cleanup and refactoring of (rule|formatter)Loader (#3412)

Commit:
Added no-dynamic-delete rule (#3573)

M0ns1gn0r committed:
Override defaultSeverity defined in extended configs  (#3449)

Commit:
[new-rule] newline-per-chained-call (#3278)

ajafff committed:
Inherit defaultSeverity and apply it to preceding base configs (#3530)

rwaskiewicz committed:
[docs]Updated no-irregular-whitespace documentation (#3627)

adidahiya committed:
Prepare release 5.9.0 (#3634)

adidahiya committed:
Disable typeof-compare in tslint:recommended (#3639)

adidahiya committed:
Resolve custom paths when loading rules (#3640)

adidahiya committed:
Prepare release 5.9.1

Shinigami92 committed:
added code-examples to no-sparse-arrays, no-unnecessary-callback-wrapper and object-literal-sort-keys

Shinigami92 committed:
Update noUnnecessaryCallbackWrapperRule.ts

Accidentally committed formatting that VSCode did, has nothing to do with the commit

Shinigami92 committed:
added code-examples to only-arrow-functions, prefer-template and typedef

Shinigami92 committed:
more examples
fixed vscode's formatting

aervin commented on src/rules/arrowReturnShorthandRule.ts:
```diff
@@ -41,6 +41,40 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Enforces to use shorthand for return.",
```
Maybe "Enforces usage of the shorthand return syntax when an arrow function's body does **not** span multiple lines."

aervin commented on src/rules/arrowReturnShorthandRule.ts:
```diff
@@ -41,6 +41,40 @@ export class Rule extends Lint.Rules.AbstractRule {
+               `,
+            },
+            {
+                description: "Enforces to use shorthand for return also for multiline.",
```
"Enforces usage of the shorthand return syntax even when an arrow function's body spans multiple lines."

aervin commented on src/rules/noEmptyInterfaceRule.ts:
```diff
@@ -30,6 +30,22 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: true,
+        codeExamples: [
+            {
+                description: "Disallows an empty interface.",
```
"Disallows empty interfaces."

aervin commented on src/rules/noSparseArraysRule.ts:
```diff
@@ -31,6 +31,20 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Disallow sparse arrays",
```
Disallows*

aervin commented on src/rules/noUnnecessaryCallbackWrapperRule.ts:
```diff
@@ -32,6 +32,24 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Disallow unnecessary callback wrappers",
```
Disallows*

aervin commented on src/rules/objectLiteralSortKeysRule.ts:
```diff
@@ -61,6 +61,58 @@ export class Rule extends Lint.Rules.OptionallyTypedRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Forces to sort keys in an object by alphabetical order",
```
Requires that an object literal's keys be sorted alphabetically.

aervin commented on src/rules/objectLiteralSortKeysRule.ts:
```diff
@@ -61,6 +61,58 @@ export class Rule extends Lint.Rules.OptionallyTypedRule {
+                `,
+            },
+            {
+                description: "Forces to sort keys in an object by alphabetical order",
```
I think this still needs updated.

aervin commented on src/rules/onlyArrowFunctionsRule.ts:
```diff
@@ -47,6 +47,28 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Disallow functions with the function keyword",
```
Disallows*

aervin commented on src/rules/preferTemplateRule.ts:
```diff
@@ -40,6 +40,41 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Forces to use template-strings",
```
"Enforces the use of template strings whenever possible."

aervin commented on src/rules/preferTemplateRule.ts:
```diff
@@ -40,6 +40,41 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Forces to use template-strings but allows up to one concatenation",
```
"Enforces the use of template strings, but allows up to one concatenation."

aervin commented on src/rules/radixRule.ts:
```diff
@@ -34,6 +34,26 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Requires the radix parameter to be specified when calling `parseInt`.",
```
"Requires the inclusion of the radix parameter when calling `parseInt`."

aervin commented on src/rules/switchDefaultRule.ts:
```diff
@@ -30,6 +30,39 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Require a `default` case in `switch` statements.",
```
Requires*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: true,
+        codeExamples: [
+            {
+                description: "Requires type definitions for call-signature",
```
call signatures*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for arrow-call-signature",
```
arrow call signatures*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for parameter",
```
parameters*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for arrow-parameter",
```
arrow function parameters*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for property-declaration",
```
property declarations*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for variable-declaration",
```
variable declarations

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for member-variable-declaration",
```
member variable declarations*

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for object-destructuring",
```
"Requires type definitions when destructuring objects."

aervin commented on src/rules/typedefRule.ts:
```diff
@@ -90,6 +90,146 @@ export class Rule extends Lint.Rules.AbstractRule {
+                `,
+            },
+            {
+                description: "Requires type definitions for array-destructuring",
```
"Requires type definitions when destructuring arrays."

aervin commented on src/rules/useIsnanRule.ts:
```diff
@@ -33,6 +33,24 @@ export class Rule extends Lint.Rules.AbstractRule {
         typescriptOnly: false,
+        codeExamples: [
+            {
+                description: "Enforces use of the `isNaN()`.",
```
How about "Enforces usage of `isNan()`."

aervin reviewed (changes requested):
Great work @Shinigami92! I just left some suggestions for the descriptions. Some of the suggestions are purely stylistic. Ignore those if you like.

Shinigami92 committed:
Improve code-example descriptions

Shinigami92 commented on src/rules/objectLiteralSortKeysRule.ts:
```diff
@@ -61,6 +61,59 @@ export class Rule extends Lint.Rules.OptionallyTypedRule {
+                `,
+            },
+            {
+                description: Lint.Utils.dedent`Requires that an object literal's keys be sorted by interface-definition.
```
Still not yet satisfying
L84 and L85

BridgeAR committed:
Remove buffers `noAssert` argument (#3770)

The support for the `noAssert` argument dropped in the upcoming
Node.js version 10.x and the argument should be removed therefore.

cwgorman committed:
Removes trailing whitespace from file-header (#3802)

Commit:
Update index.md (#3773)

JSON cannot have dangling comma

a-xin committed:
Improve documentation of cli flag --project (#3703)

The current documentation of the cli flag `--project` is a little confusing as to which the path of a file or the path of a directory can be specified.

As both are valid options, updating the documentation to make it clear that either the path to a `tsconfig.json` file or the path to a directory containing a `tsconfig.json` file can be specified.

sehrope committed:
no-unbound-method: Fix description typo (#3732)

Commit:
Fixed "manging" typo in no-this-assignment (#3673)

rafaelss95 committed:
docs(changelog): fix wrong description (#3659)

Commit:
Update for docs/usage/configuration/index.md (#3661)

Added fix for text issue

Commit:
Added 2 (+/-1) sentence rationales for 2/3 of rules (#3734)

* Added a couple extra docs lines to align

* Added a couple extra docs lines to arrow-return-shorthand

* Added a couple extra docs lines to await-promise

* Added a couple extra docs lines to binary-expression-operand-order

* Added a couple extra docs to class-name

* Added a couple extra docs to completed-docs

* Added another doc line to cyclomatic-complexity

* Switch link to https in eofline

* Added a couple extra docs to for-in

* Added a couple extra docs to member-access

* Added a couple extra docs to member-ordering

* Added a couple extra docs to no-any

* Added a couple extra docs to no-boolean-literal-compare

* Added a couple extra docs to no-consecutive-blank-lines

* Added a couple extra docs to no-dynamic-delete

* Added a couple extra docs to no-floating-promises

* Added a couple extra docs to no-inferred-empty-object-type

* Added a brief rationale to no-invalid-template-strings-rule

* Typo correction on no-magic-numbers

* Added a couple extra docs to no-misused-new

* Added a couple extra docs to no-non-null-assertion

* Added a couple extra docs to no-null-keyword

* Added a couple extra docs to no-parameter-properties

* Added a couple extra docs to no-shadowed-variable

* Added a couple extra docs to no-string-throw

* Added a couple extra docs to no-this-assignment

* Added a couple extra docs to no-unbound-method

* Added a couple extra docs to no-unnecessary-callback-wrapper

* Added a couple extra docs to no-unnecessary-initializer

* Added a couple extra docs to no-unsafe-any

* Added a couple extra docs to no-unused-variable

* Added a couple extra docs to no-var-keyword

* Added a couple extra docs to no-var-requires

* Added a couple extra docs to no-void-expression

* Added a couple extra docs to number-literal-format

* Missing dedent in no-non-null-assertion

LucasSloan committed:
Build tslint with TypeScript 2.7.2. (#3819)

* Build tslint with TypeScript 2.7.2.

Updates the package.json, fixes some compiler issues where some typescript APIs changed, and makes 2 rules work under 2.7.2.

* Fixes from code review.

Fixed a lint issue (' => ").
Actually checked for undefined, instead of just asserting.

blair committed:
promise-function-async: add options to specify function types to check. (#3807)

This is useful if you only want your function and method declarations
to warn, or in other words, one doesn't want each arrow function that
is passed to Promise#then() or Promise#catch() to be async.

Commit:
Used bang in file-header (#3741)

jishi9 committed:
Clarified lint message "All imports are unused." (#3831)

Changed to: "All imports on this line are unused"

Commit:
Updated max-line-length ignore-pattern example (#3834)

The example uses:

```
class [a-zA-Z] implements 
```

but this only matches a class that has a single character as name. Even if it is an example, the rule should accept more than one character for the class name. I struggled with this ignore pattern in a project, because I didn't read the pattern itself completely. Therefore I want to give this feedback, such that other people don't have to search for the reason of a failing rule, like me.

markvincze committed:
Add the option allow-empty-functions in no-empty (#3624)

Commit:
[docs] adds optional capability to provide code examples in rules' metadata (#3602)

* codeExample support init implementation

* dedent/markdown errors need revisting

* updates to new code example scheme as per ajafff

* reverts prettier style updates in curly

* curly docs he -> they

* ICodeExample config type any -> string

* Code example blocks now color-coded according to pass/fail (grn/red)

* updates to buildDocs

* gemfile.lock??; removes unnecessary html/scss

* removes site dist

* moving code examples to dedicated dir to avoid bloat in rule files

* removing _site dir

suchanlee committed:
Fix build (#3853)

* fix non-no-unused-variable tests

* Fix no-unused-variables tests and modify logic in noUnusedVariableRule for TS2.8 support

Logic had to be modified because starting from TS2.8, when all imports in an import node are not used,
TS emits only 1 diagnostic object for the whole line as opposed to the previous behavior of outputting
a diagnostic with kind == 6192 (UnusedKind.VARIABLE_OR_PARAMETER) for every unused import.

* fix lint issue

* hack/fix check-parameters test for 2.9-dev

* add builds for ts 2.7 and 2.8

* fix tests that broke from diagnosis codes changing in 2.9

Commit:
Removed *nix-only comment from package.json (#3858)

It feels self-evident that `lint:global` as a task would lint everything with the global TSLint install.

tanniiz committed:
[new-fixer]: Add space before comment message fixer (#3845)

* [new-fixer]: Add space before comment message fixer

* Add space-before-comment test

Commit:
Added rule+file-specific error messages to rule crashes (#3836)

* Added rule+file-specific error messages to rule crashes

It's not clear where tests for this should live. Would you like me to add some, and if so where?

* Added endline; no more unnecessary duplication

felipeissa committed:
Adding documentation example to deprecationRule (#3848)

This adds documentation example to the deprecation rule.

304NotModified committed:
Improved "complete docs" docs (#3823)

BB9z committed:
[docs] no-string-throw: add missing examples. (#3701)

* [docs] no-string-throw: add missing examples.

* [docs] no-string-throw: examples for rationale.

* Code example section for no-string-throw.

mwld committed:
correct name and link for "object-literal-sort-keys" in changelog (#3715)

ethanresnick committed:
no-unbound-method: add another exception (#3610)

Reading a property with square brackets, though rare, should be allowed because it won't be effected by an unbound this.

rwaskiewicz committed:
Fix completed-docs rule (#3868)

- Fixes building of master
- Add escaping to backticks introduced in #3823

rwaskiewicz committed:
[new-rule]prefer-while rule (#3750)

* prefer-while rule

Implements prefer-while rule in place of a for loop without an initializer and incrementer

* renamed failure object creation method

* fix json test file indentatons

* Address review comments

 
- Use isForStatement from tsutils
- Update failure string to be more descriptive
- Updated test messages

* remove uneccessary type assertion

- assertion now uneccessary with using isForStatement from tsutils

* Add rationale field to prefer-while rule metadata

* Use 'let' instead of 'const' in test

* Update rule description & failure string

- Updated the description and failure string of prefer-while
- Updated tests accordingly

* Coalesce test cases

- Combine test cases for 'bad' for loops and case for 'good' for loops
- Remove unneeded files
- Add comments to test cases

* Add additional test cases

- Added test case with incrmentor in body of loop
- Added test case with += incrementor

* Add code examples

- Added code examples for prefer-while rule

Shinigami92 committed:
Merge branch 'docs/code-examples' of https://github.com/aervin/tslint into docs/code-examples

Shinigami92 committed:
move codeExamples into new directory

aervin commented:
@Shinigami92 Looks awesome! You should open this on the primary TSLint repo so it can be reviewed there.

Shinigami92 committed:
Merge branch 'master' of https://github.com/palantir/tslint into docs/code-examples

Shinigami92 commented:
@aervin Pull requested on https://github.com/palantir/tslint/pull/3869

OliverJAsh committed:
Improve "no valid rules" error message to include file type (#3729)

* Improve "no valid rules" error message to include file type

* Lint

Commit:
Excluded native JSX elements from no-unsafe-any (#3699)

* Excluded native JSX elements from no-unsafe-any

* Lint fix: double-apostrophe

Interesting how TypeScript auto-added single-apostrophes...

* ...almost working..

* Split no-unsafe-any into separate tests

* Added test cases for fragments

* Added case for TS2.1 and fragments

jkillian committed:
Prepare release 5.10.0 (#3871)

* Prepare release 5.10.0

bowenni committed:
Fix Copyright: @license JSDoc tag was missing (#3879)

pe8ter committed:
Fix some typos in `no-floating-promises` rule docs. (#3886)

mesaugat committed:
Fix typo (#3898)

```diff
+ for
- or
```

Commit:
Fix link to Configuration page (#3891)

rwaskiewicz committed:
[docs] Fix prefer-while docs (#3888)

- Updated docs to make sense semantically (use vars that should exist in the scope of each for-loop)
- Updated tests with same issue

keradus committed:
CS: test/rules/no-empty/default/test.ts.lint - fix missing newline at end of file (#3896)

keradus committed:
no-empty: fix lack of methods handling (#3897)

* no-empty: ensure rule works for typed functions

* no-empty: prove lack of methods handling

* fix

liuxh0 committed:
Fix typo (#3910)

rkirov committed:
Turn on strictPropertyInitialization for src/ and test/. (#3924)

Some Walker fields had to be initialized with inline initializers.

Commit:
init patch; using Object.is to diff between pos/neg zero (#3903)

Commit:
bugfix - no-use-before-declare - exception for destructuring w/ rename (#3876)

* ignoring identifiers whose parent is a binding element

* check that ignored identifier is propname of binding element

* additional tests

Commit:
Added fixer for switch-final-break (#3615)

mertdeg2 committed:
Handle tsconfig.json errors without using JSON.stringify (#3908)

JSON.stringify fails in some cases because its fails to convert circular structure to string

jkillian committed:
Deprecate no-unused-variable rule for typescript@next; also fix runtime error (#3919)

* fix no-unused-variable-rule for typescript@next

* fix ignore pattern test file as well

* fix tests

* deprecate noUnusedVariableRule for TS 2.9+

* use semver for cleaner comparison check

jkillian committed:
don't run no-unused-variable tests when testing against version of TS it's deprecated for (#3958)

NaridaL committed:
Use Buffer.allocUnsafe instead of the deprecated new Buffer() (#3985)

(node:16148) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
allocUnsafe should be fine because in both cases, the buffer is immediately filled and thrown out.

NaridaL committed:
Implemented fixer for member-ordering and added corresponding tests. (#3935)

* Implemented fixer for member-ordering and added corresponding tests.

* Improved node boundary calculation so trivia remains attached to the correct nodes.

* minor comment fixes.

david-cannady committed:
Radix rule checks Number.parseInt(), window.Number.parseInt(), global.Number.parseInt() (#3901)

euclio committed:
[New rule] add fileNameCasing rule (#3978)

* add fileNameCasing rule

* use camel case name for tslintCli

* refactor casing check into helper function

* use stylized name for casing in failure message

saberduck committed:
no-implicit-dependencies should include peer dependencies even with "dev" option set (#3875)

NaridaL committed:
Output +/- on diff so added/removed empty lines are visible. (#3973)

ajcrites committed:
[new-rule-option] [no-implicit-dependencies] - add whitelist (#3839) (#3979)

* feature - no-implicit-dependencies - add whitelist (#3839)

This allows you to specify a whitelist of modules that will not be part of package.json such as project-relative aliases

* feature - no-implicit-dependencies - add whitelist (#3839)

Use Set for whitelist package name comparison instead of Array.

pablobirukov committed:
Fix CI: no-implicit-dependencies test failure; typescript@next failure (#4019)

* Fix CI, which became red after consequent merge #3979, #3875

* Upgrade typescript@2.9.2, tsutils@2.27.2

* Remove unneded type assertion after typescript upgrade

* Add runtime check for 'ts.unescapeIdentifier()' presence

Shinigami92 committed:
changed year in code-examples copyrights

Shinigami92 committed:
Merge branch 'master' into docs/code-examples

Shinigami92 committed:
Merge branch 'master' into docs/code-examples

The GitHub GUI is a bit confusing

Shinigami92 committed:
wrong import
